packages/babel-core/src/transformation/block-hoist-plugin.ts

```diff
@@ -17,14 +19,14 @@ export default function loadBlockHoistPlugin(): Plugin {
 
   return LOADED_PLUGIN;
 }
-function priority(bodyNode) {
+function priority(bodyNode: Statement & { _blockHoist?: number | boolean }) {
```

Comment:
```suggestion
function priority(bodyNode: Statement & { _blockHoist?: number | true }) {
```

and you should remove casting of `priority as number`